.github/workflows/issue_triage.yml

```diff
@@ -0,0 +1,88 @@
+            
+      - name: Check Labels for Proper Triaging
+        id: check-labels
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const issue = await github.rest.issues.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number
+            });
+            const labels = issue.data.labels.map(label => label.name);
+            const requiredLabels = {
+              types: ['bug', 'enhancement', 'good first issue'],
+              impacts: ['Impact: High', 'Impact: Medium', 'Impact: Low'],
+              work: ['Work: High', 'Work: Medium', 'Work: Low'],
+              priorities: ['Priority: Essential', 'Priority: Important', 'Priority: Nice-to-have']
+            };
+            const hasTypeLabel = labels.some(label => requiredLabels.types.includes(label));
+            const hasImpactLabel = labels.some(label => requiredLabels.impacts.includes(label));
+            const hasWorkLabel = labels.some(label => requiredLabels.work.includes(label));
+            const hasPriorityLabel = labels.some(label => requiredLabels.priorities.includes(label));
+            const needsTriage = !(hasTypeLabel && hasImpactLabel && hasWorkLabel && hasPriorityLabel);
+            core.setOutput('needs-triage', needsTriage);
+
+      - name: Manage Needs Triage Label
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const needsTriage = '${{ steps.check-labels.outputs.needs-triage }}' === 'true';
+            if (needsTriage) {
+              await github.rest.issues.addLabels({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.issue.number,
+                labels: ['needs-triage']
+              });
+            } else {
+              try {
+                await github.rest.issues.removeLabel({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  issue_number: context.issue.number,
+                  name: 'needs-triage'
+                });
+              } catch (error) {
+                if (error.status !== 404) {
+                  throw error;
+                }
+              }
+            }
+
+      - name: Verify Project Assignment
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const projects = await github.rest.issues.listForRepo({
+              owner: context.repo.owner,
+              repo: context.repo.repo
+            });
+            const assignedProjects = projects.data.filter(project => project.state === 'open');
+            if (assignedProjects.length !== 1) {
+              console.warn(`Issue must be assigned to exactly one open project. Found: ${assignedProjects.length}`);
+            } else {
+              console.log(`Valid project assignment: ${assignedProjects[0].name}`);
+            }
```

Comment:
For this step, I'm wondering if we need to post a comment so that the triager is reminded to assign a project. What do you think?

Reply:
I think it would be good idea to post a comment because it would make the workflow action more clear.

Reply:
I was trying out this as it can be seen https://github.com/Mayank77maruti/oppia-android/issues/12 this part is running several tmes. i am doing this currently on basis of label when label are added other than `needs-triage` label the workflow will run project assigned workflow, but in my case the three labels are taken as individual the workflow runs thrice.

Reply:
This sounds good, thanks!